// twisted/dist/constants/regions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Regions;
(function (Regions) {
    Regions["BRAZIL"] = "BR1";
    Regions["EU_EAST"] = "EUN1";
    Regions["EU_WEST"] = "EUW1";
    Regions["KOREA"] = "KR";
    Regions["LAT_NORTH"] = "LA1";
    Regions["LAT_SOUTH"] = "LA2";
    Regions["AMERICA_NORTH"] = "NA1";
    Regions["OCEANIA"] = "OC1";
    Regions["TURKEY"] = "TR1";
    Regions["RUSSIA"] = "RU";
    Regions["JAPAN"] = "JP1";
    Regions["PBE"] = "PBE1";
})(Regions = exports.Regions || (exports.Regions = {}));
var TftRegions;
(function (TftRegions) {
    TftRegions["EUROPE"] = "europe";
    TftRegions["ASIA"] = "asia";
    TftRegions["AMERICAS"] = "americas";
})(TftRegions = exports.TftRegions || (exports.TftRegions = {}));
function regionToTftRegions(region) {
    switch (region) {
        // America
        case Regions.AMERICA_NORTH:
        case Regions.BRAZIL:
        case Regions.LAT_NORTH:
        case Regions.LAT_SOUTH:
        case Regions.OCEANIA:
            return TftRegions.AMERICAS;
        // Europe
        case Regions.EU_EAST:
        case Regions.EU_WEST:
        case Regions.TURKEY:
        case Regions.RUSSIA:
            return TftRegions.EUROPE;
        // Asia
        case Regions.JAPAN:
        case Regions.KOREA:
            return TftRegions.ASIA;
    }
    throw new Error(`Unexpected region: ${region}`);
}
exports.regionToTftRegions = regionToTftRegions;
